cEP-0019.md

```diff
@@ -0,0 +1,454 @@
+   time. In particular, one will always see the review comment that is under
+   his PR first.
+
+2. Meta-review by maintainers. For maintainers, review comments after
+   merge/close also need meta-reviews. Those meta-reviews will reflect the
+   accuracy of the review comments.
+
+##### Display
+
+Review comments that need meta-review will be displayed on a separate column
+on gh-board. There are three approaches:
+
+1. The worst case is that the tracking stores in the json which
+   PR do not have meta-reviews, and gh-board shows only a list of those PRs
+   (and the user must then find the reviews in GitHub PR web app, and do
+   meta-review there).
+
+2. An improvement is to store the identifier of the PR comments which
+   do not have meta-reviews and gh-board links to the PR review comment
+   in GitHub PR web app.
+
+3. In the ultimate version, gh-board would show the review comments.
+   In the future, people could do meta-reviews directly on gh-board. This also
+   provides an approach to doing meta-reviews anonymously.
+
+#### Performance Requirements
+
+Currently,
+[fetch-issues.js](https://github.com/coala/gh-board/blob/46f1ad5ce83d5d72aa95ea00703b92c2fc38b2dc/script/fetch-issues.js)
+pre-fetches issues and pull requests to speed up gh-board performance and
+save users' API hits. Meta-reviews are also to be fetched during this procedure.
+
+#### Other Features
+
+The GSoC project provides a blueprint for future enhancement. Additional
+features could be implemented to make the meta-review system more
+sophisticated:
+
+1. A basic ranking list will be built first. In the best case, we should
+   offer an overall ranking list as well as a recent ranking list.
+
+2. Offer an option for users to do meta-reviews directly on gh-board. This
+   might be a bad idea - people shouldn't do meta-review unless they know the
+   whole PR context. This might be a good idea - in the future users can do
+   meta-reviews on gh-board anonymously (there is no way for people to do
+   anonymous meta-reviews on GitHub page).
+
+3. Display visitor’s own meta-reviews if the visitor logs in. People may want
+   to see meta-reviews they have done before.
+
+4. Provide notifications of meta-reviews to a user if he receives any
+   meta-review. Suppose I ask for someone's agreement and he uses 👍 to
+   show his agreement. But I wouldn't receive any notification. I have to
+   check that page regularly. gh-board is a nice place to provide notifications
+   of meta-reviews.
+
+5. There is a potential for reverted commits to be meta-reviewed. A reverted
+   commit should trigger a new meta-review of the original PR, as time may
+   have proved that the merge was incorrect, or maybe the merge was a
+   self-aware time-sensitive workaround, and is no longer needed.
+
+6. The UI of gh-board could have factual aggregated statistical information
+   like how many of each emoji a person gave/received, again with a legend
```

Comment:
a person has given/received

Present Perfect because that person might give more emoji later (so, the process of "giving emoji" is not finished)